music21/metadata/__init__.py

```diff
@@ -870,35 +1369,25 @@ def filePath(self) -> t.Optional[str]:
         if self.fileInfo.path:
             return str(self.fileInfo.path)
 
-    @filePath.setter
-    def filePath(self, value: t.Union[str, Text]) -> None:
-        self.fileInfo.path = Text(value)
-
```

Comment:
as above.  -- if these setters aren't needed, maybe the getters aren't either?  (I realize now that I added these in the middle of this PR, so maybe they should all go and just be stored as standard metadata?  Just to make sure that the corresponding Score properties still work)

Reply:
AH!  I get it, the getters are remaining just for docs, but are unnecessary.  Ok.

Reply:
Actually, I think these will fool static type checkers into thinking that the property is not settable.  Will need to test.

Reply:
Here's the problem: __settattr__ is rude. If you implement it, none of your property setters will ever be called again. (__getattr__ is not rude like that). So I had to move this code into __setattr__. You'll notice that all the property setters are gone (e.g. @composer.setter et al).